Linking fails when using Allegro

For some reason I can't compile LibreSprite, ninja fails every time while linking.
```
git clone --recursive https://github.com/LibreSprite/LibreSprite
mkdir build
cd build
cmake -DUSE_SDL2_BACKEND=off -DUSE_ALLEG4_BACKEND=on -G Ninja ..
ninja libresprite
```

<details>
<summary>Output</summary>

```
[833/833] Linking CXX executable bin/libresprite
FAILED: bin/libresprite 
: && /usr/bin/c++ -std=c++14 -Wall -Wno-switch -std=gnu++11 -O2 -g -DNDEBUG  src/CMakeFiles/libresprite.dir/main/main.cpp.o src/CMakeFiles/libresprite.dir/main/xpm_icon.c.o -o bin/libresprite  -Wl,--whole-archive  lib/libapp-lib.a  -lm  -lpthread  /usr/lib/libSM.so  /usr/lib/libICE.so  /usr/lib/libX11.so  /usr/lib/libXext.so  /usr/lib/libXext.so  /usr/lib/libXcursor.so  /usr/lib/libXpm.so  /usr/lib/libXxf86vm.so  -lpthread  lib/libcfg-lib.a  lib/libcss-lib.a  lib/libfilters-lib.a  lib/libflic-lib.a  lib/libnet-lib.a  /usr/lib/libcurl.so  lib/librender-lib.a  lib/libdoc-lib.a  lib/libfixmath-lib.a  lib/libscript-lib.a  lib/libduktape.a  lib/libui-lib.a  lib/libclip.a  -lxcb  -lpthread  lib/libshe.a  lib/libbase-lib.a  lib/libobs.a  lib/libmodp_b64.a  /usr/lib/libdl.so  lib/libgfx-lib.a  lib/libloadpng.a  lib/liballeg.a  -lm  -lpthread  /usr/lib/libSM.so  /usr/lib/libICE.so  /usr/lib/libX11.so  /usr/lib/libXext.so  /usr/lib/libXcursor.so  /usr/lib/libXpm.so  /usr/lib/librt.so  /usr/lib/libXxf86vm.so  /usr/lib/libpixman-1.so  lib/libundo.a  -Wl,--no-whole-archive  /usr/lib/libtinyxml.so  /usr/lib/libjpeg.so  /usr/lib/libgif.so  /usr/lib/libpng.so  /usr/lib/libz.so  /usr/lib/liblua5.4.so  -lm  /usr/lib/libfreetype.so && :
/usr/bin/ld: lib/libui-lib.a(entry.cpp.o): in function `ui::Entry::onProcessMessage(ui::Message*)':
/home/marco/Workspace/LibreSprite/src/ui/entry.cpp:214: undefined reference to `she::set_input_rect(gfx::RectT<int> const&)'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

</details>

_Originally posted by @Gaarco in https://github.com/LibreSprite/LibreSprite/issues/289#issuecomment-950275372_

--- a/src/she/alleg4/she.cpp
+++ b/src/she/alleg4/she.cpp
@@ -1,5 +1,6 @@
 // SHE library
 // Copyright (C) 2012-2016  David Capello
+// Copyright (C) 2021       LibreSprite contributors
 //
 // This file is released under the terms of the MIT license.
 // Read LICENSE.txt for more information.

